lib/system.nim

```diff
@@ -3043,3 +3043,6 @@ when defined(nimDebugUtils):
     {.define(nimCompilerDebug).}
     n
     {.undef(nimCompilerDebug).}
+
+# Overload for not nil
+proc `not`*(a: typedesc[ref], b: typeof(nil)): typedesc {.magic: "TypeTrait".}
```

Comment:
```suggestion
proc `not`*(a: typedesc[ptr or ref], b: typeof(nil)): typedesc {.magic: "TypeTrait".}
```
So that `T not nil` also works with pointer types. Also, perhaps it'd make sense to move the procedure together with the other type constructor procedures (`not`, `and`, and `or`)?

Reply:
The other type constructor procedures appear before the definition of `typeof` which this one uses. I could move those or typeof around but maybe that's better left for another PR to keep the diff cleaner

Reply:
Oh, I hadn't considered that. The `system` module really needs to be cleaned up and reordered at some point, but I think it's fine if `not` stays where it is for now.

Reply:
Did you encounter problems with using `ptr or ref`?